content/blog/open-sourcing-workshops-on-demand-part2-deploying-the-backend.md

```diff
@@ -0,0 +1,389 @@
+
+### For private based Workshops-on-Demand (private backend + private workshops on top of default public backend and notebooks)
+
+T﻿he principle remains somehow similar with a few differences explained below.
+
+* Y﻿ou will start by forking the following public private [repo](https://github.com/Workshops-on-Demand/wod-private.git) on github under your own github account (we will refer to it as `Account`)
+* Next, you clone the forked repo
+
+```shellsession
+install$ git clone https://github.com/Account/wod-private.git wod-private
+install$ cd $HOME/wod-private/ansible/group_vars
+```
+
+* Please edit the `all.yml` and `<groupname>` files to customize your setup. T﻿his variable `<groupname>`defines possible backend server in your environement. By default, the project comes with a sample working file named `production` in `ansible/group-vars`. But you could have multiple. In our case, we have defined `sandbox`, `test`, `staging` and several `production` files, all defining a different backend environment. These files will be used to override the default values specified by the public version delivered as part of the default public installation.
+* Commit and push changes to your repo
+* Create an `install.priv` file located in `install` directory when using a private repo (consider looking at [install.repo](https://github.com/Workshops-on-Demand/wod-backend/blob/main/install/install.repo) file for better understanding of the variables.
+
+  * Define the WODPRIVREPO and WODPRIVBRANCH variables as follows:
+
+    * WODPRIVBRANCH="main"
+    * WODPRIVREPO="git@github.com:Account/Private-Repo.git wod-private"
+
+**Note: When using a token**
+
+Please refer to the following [url](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) to generate a `token` file in `install` directory of wod-backend:
+
+* edit the `install.priv` file located in `install` directory of wod-backend:
+
+  * create line before variable declaration: ``token=`cat $EXEPATH/token` ``
+  * use the token in the url WODPRIVREPO="git clone https://user:$token@github.com/Account/wod-private.git wod-private"
+
+Y﻿ou are now ready to perform the installation again to support a private repository. 
+
+```shellsession
+cd $HOME/wod-backend/install
+sudo ./install.sh -t backend -g staging -b jup.example.net -f notebooks.example.io -a api.example.io -e notebooks.example.io -s sender@example.io
+```
+
+Please note that this setup phase can be conccurent with the public setup phase. Indeed, the install script should detect the presence of the private repository owing to the presence of the install.priv file. It will automatically adjust the different scripts and variables to add the relevant content. It will actually overload some of the variables with private ones.
```

Comment:
At the end of this paragraph, do you mean "overload" or "overwrite"?

Reply:
overlaod